resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/MultiValuedCollectionParamConverter.java

```diff
@@ -0,0 +1,139 @@
+package org.jboss.resteasy.plugins.providers;
+
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.Collection;
+
+import javax.annotation.Priority;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.core.StringParameterInjector;
+import org.jboss.resteasy.resteasy_jaxrs.i18n.Messages;
+
+/**
+ * @author Marek Kopecky mkopecky@redhat.com
+ * @author Ron Sigal rsigal@redhat.com
+ */
+@Provider
+@Priority(Integer.MAX_VALUE)
+public class MultiValuedCollectionParamConverter implements ParamConverter<Collection<?>> 
```

Comment:
Why not create an Abstract class`MultiValuedParamConverter` to provide base impl for construtor with `StringParameterInjector` and `separator` and impl for `getSeparatorChar()` and `stringify()`.
It will be shared betwen `MultiValuedCollectionParamConverter` and `MultiValuedArrayParamConverter`.
WDYT?

Reply:
@NicoNes, yeah, ok. I used Arrays.asList().